src/client-side-metrics/gcp-metrics-handler.ts

```diff
@@ -0,0 +1,189 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  IMetricsHandler,
+  OnAttemptCompleteMetrics,
+  OnOperationCompleteMetrics,
+} from './metrics-handler';
+import * as Resources from '@opentelemetry/resources';
+import * as ResourceUtil from '@google-cloud/opentelemetry-resource-util';
+import {MetricExporter} from '@google-cloud/opentelemetry-cloud-monitoring-exporter';
+import {
+  OnAttemptCompleteAttributes,
+  OnOperationCompleteAttributes,
+} from '../../common/client-side-metrics-attributes';
+const {
+  MeterProvider,
+  Histogram,
+  Counter,
+  PeriodicExportingMetricReader,
+} = require('@opentelemetry/sdk-metrics');
+
+/**
+ * A collection of OpenTelemetry metric instruments used to record
+ * Bigtable client-side metrics.
+ */
+interface Metrics {
+  operationLatencies: typeof Histogram;
+  attemptLatencies: typeof Histogram;
+  retryCount: typeof Counter;
+  applicationBlockingLatencies: typeof Histogram;
+  firstResponseLatencies: typeof Histogram;
+  serverLatencies: typeof Histogram;
+  connectivityErrorCount: typeof Histogram;
+  clientBlockingLatencies: typeof Histogram;
+}
+
+/**
+ * A metrics handler implementation that uses OpenTelemetry to export metrics to Google Cloud Monitoring.
+ * This handler records metrics such as operation latency, attempt latency, retry count, and more,
+ * associating them with relevant attributes for detailed analysis in Cloud Monitoring.
+ */
+export class GCPMetricsHandler implements IMetricsHandler {
+  private initialized = false;
+  private otelMetrics?: Metrics;
+
+  /**
+   * Initializes the OpenTelemetry metrics instruments if they haven't been already.
+   * Creates and registers metric instruments (histograms and counters) for various Bigtable client metrics.
+   * Sets up a MeterProvider and configures a PeriodicExportingMetricReader for exporting metrics to Cloud Monitoring.
+   * @param {string} [projectId] The Google Cloud project ID. Used for metric export. If not provided, it will attempt to detect it from the environment.
+   */
+  private initialize(projectId?: string) {
+    if (!this.initialized) {
+      this.initialized = true;
+      // Use MeterProvider provided by user
+      // If MeterProvider was not provided then use the default meter provider.
+      const meterProvider = new MeterProvider({
+        // This is the default meter provider
+        // Create a resource. Fill the `service.*` attributes in with real values for your service.
+        // GcpDetectorSync will add in resource information about the current environment if you are
+        // running on GCP. These resource attributes will be translated to a specific GCP monitored
+        // resource if running on GCP. Otherwise, metrics will be sent with monitored resource
+        // `generic_task`.
```

Comment:
are these comments intentional?

Reply:
This comment was a copy/paste from the code copied from the npm registry used for exporting metrics to GCP. I just removed it.